src/Symfony/Component/Workflow/Dumper/PlantUmlDumper.php

```diff
@@ -0,0 +1,171 @@
+                'BackgroundColor' => '#ffffff',
+                'BorderColor' => '#3887C6',
+            ),
+        ),
+    );
+
+    private $transitionType = self::DEFAULT_TRANSITION_TYPE;
+
+    public function __construct(string $transitionType = null)
+    {
+        if (!in_array($transitionType, self::TRANSITION_TYPES)) {
+            throw new InvalidArgumentException("Transition type '{$transitionType}' does not exist.");
+        }
+        $this->transitionType = $transitionType;
+    }
+
+    public function dump(Definition $definition, Marking $marking = null, array $options = array()): string
+    {
+        $options = array_replace_recursive(self::DEFAULT_OPTIONS, $options);
+        $code = $this->initialize($options);
+        foreach ($definition->getPlaces() as $place) {
+            $code[] =
+                "state {$place}".
+                ($definition->getInitialPlace() === $place ? ' <<'.self::INITIAL.'>>' : '').
+                ($marking && $marking->has($place) ? ' <<'.self::MARKED.'>>' : '');
+        }
+        if ($this->isSquareTransitionType()) {
+            foreach ($definition->getTransitions() as $transition) {
+                $code[] = "agent {$transition->getName()}";
+            }
+        }
+        foreach ($definition->getTransitions() as $transition) {
+            foreach ($transition->getFroms() as $from) {
+                foreach ($transition->getTos() as $to) {
+                    if ($this->isSquareTransitionType()) {
+                        $lines = array(
+                            "{$from} --> {$transition->getName()}",
+                            "{$transition->getName()} --> {$to}",
+                        );
+                        foreach ($lines as $line) {
+                            if (!in_array($line, $code)) {
+                                $code[] = $line;
+                            }
+                        }
+                    } else {
+                        $code[] = "{$from} --> {$to}: {$transition->getName()}";
+                    }
+                }
+            }
+        }
+
+        return $this->startPuml($options).$this->getLines($code).$this->endPuml($options);
+    }
+
+    private function isSquareTransitionType(): bool
+    {
+        return static::SQUARE_TRANSITION_TYPE === $this->transitionType;
+    }
+
+    private function startPuml(array $options): string
+    {
+        $start = '@startuml'.PHP_EOL;
+
+        if ($this->isSquareTransitionType()) {
+            $start .= 'allow_mixing'.PHP_EOL;
+        }
+
+        if ($options['nofooter'] ?? false) {
+            return $start;
+        }
+
+        return $start.static::SYMFONY_LOGO.PHP_EOL;
+    }
+
+    private function endPuml(array $options): string
+    {
+        $end = PHP_EOL.'@enduml';
+        if ($options['nofooter'] ?? false) {
+            return $end;
+        }
+
+        return PHP_EOL.'footer \nGenerated by <$sf_logo> **Workflow Component** and **PlantUML**'.$end;
```

Comment:
Are you sure about the `\n` here, which is not a newline char ?

Reply:
yes I think so, the `\n` is working as expected.